src/components/va-back-to-top/va-back-to-top.tsx

```diff
@@ -0,0 +1,78 @@
+import { Component, Element, Host, State, h } from '@stencil/core';
+import classnames from 'classnames';
+
+import 'intersection-observer';
+
+@Component({
+  tag: 'va-back-to-top',
+  styleUrl: 'va-back-to-top.css',
+  shadow: true,
+})
+export class VaBackToTop {
+  revealPixel!: HTMLSpanElement;
+
+  @State() hasHitBreakpoint = false;
+  @State() isDocked = false;
+
+  @Element() el: HTMLElement;
+
+  componentDidLoad() {
+    // Setup observer to handle docking behavior
+    const dockObserver = new IntersectionObserver(
+      entries => {
+        // The second half of the || ensures that the button
+        // stays docked even if the viewport is scrolled below the dock
+        this.isDocked =
+          entries[0].isIntersecting || entries[0].boundingClientRect.y < 0;
+      },
+      { threshold: 1.0 },
+    );
+    dockObserver.observe(this.el);
+
+    // Setup observer to handle reveal behavior
+    const revealObserver = new IntersectionObserver(entries => {
+      this.hasHitBreakpoint = entries[0].boundingClientRect.y < 0;
+    });
+    revealObserver.observe(this.revealPixel);
+  }
+
+  navigateToTop() {
+    // Focus the h1 tag on the page.
+    const el = document.querySelector('h1');
+    if (el) {
+      // Prepare the element so that it can accept focus properly.
+      el.setAttribute('tabindex', '-1');
+      el.focus();
+
+      // Cleanup the tabindex on blur.
+      el.addEventListener('blur', () => el.removeAttribute('tabindex'));
+    }
+
+    // Scroll to the top.
+    return window.scrollTo(0, 0);
+  }
+
+  render() {
+    const undockedWidth = this.el.getBoundingClientRect().width;
+    this.el.style.setProperty('--undocked-width', `${undockedWidth}px`);
```

Comment:
This ensures that when the button is revealed but not docked, its width is the same as its parent.

Reply:
Smells like a good code comment for posterity. :wink: I also recommend mentioning how we're using the `--undocked-width` in the `style` since that's not a native _thing._

Reply:
What do you mean it isn't a native thing? I'm setting it [in a standard way](https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_custom_properties#values_in_javascript), and CSS custom properties are already used elsewhere.

I'll add the code comment saying that this is what restricts the width :+1:

Reply:
Right right, what I'm saying is that `--undocked-width` is a CSS custom property, not a standard CSS property. So a bit of explanation might be good for what we're using that custom property for.

What you've got now is great. :+1: